modules/lang-groovy/src/main/java/org/elasticsearch/script/groovy/GroovyScriptEngineService.java

```diff
@@ -91,6 +92,11 @@
      */
     private final ClassLoader loader;
 
+    /**
+     * Ensures that the deprecation log entry for Groovy is only written on the first Groovy script compiled.
+     */
+    private AtomicBoolean isDeprecationLogged = new AtomicBoolean(false);
```

Comment:
I agree - would be good to log each time we have a request using a deprecated script (but not on every execution of that script :D) so that the deprecation warning gets sent back to the client